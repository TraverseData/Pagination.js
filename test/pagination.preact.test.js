
import preact from 'preact'
import render from 'preact-render-to-json'

import Pagination from '../preact/Pagination'

/** @jsx preact.h */

const wrapper = props => <ul>{props.children}</ul>
const previous = page => <li key="previous"><a href={page - 1}>Previous</a></li>
const next = page => <li key="next"><a href={page + 1}>Next</a></li>
const spacer = page => <li key={page}>...</li>
const item = (page, active) => (
    active ?
    (<li key={page}>{page}</li>) :
    (<li key={page}><a href={page}>{page}</a></li>)
)

test('1st page, 100 total, 10 per page', () => {
    const tree = render(
        <Pagination
            page={1}
            count={100}
            perPage={10}
            context={2}
            previous={previous}
            wrapper={wrapper}
            item={item}
            next={next}
            spacer={spacer}
        />)

    expect(tree).toMatchSnapshot()
})

test('3rd page, 45 total, 10 per page', () => {
    const tree = render(
        <Pagination
            page={3}
            count={45}
            perPage={10}
            context={2}
            previous={previous}
            wrapper={wrapper}
            item={item}
            next={next}
            spacer={spacer}
        />)

    expect(tree).toMatchSnapshot()
})

test('10th page, 200 total, 10 per page', () => {
    const tree = render(
        <Pagination
            page={10}
            count={200}
            perPage={10}
            context={2}
            previous={previous}
            wrapper={wrapper}
            item={item}
            next={next}
            spacer={spacer}
        />)

    expect(tree).toMatchSnapshot()
})

test('10th page, 100 total, 10 per page', () => {
    const tree = render(
        <Pagination
            page={10}
            count={100}
            perPage={10}
            context={2}
            previous={previous}
            wrapper={wrapper}
            item={item}
            next={next}
            spacer={spacer}
        />)

    expect(tree).toMatchSnapshot()
})

test('1st page, 10 total, 10 per page', () => {
    const tree = render(
        <Pagination
            page={1}
            count={10}
            perPage={10}
            context={2}
            previous={previous}
            wrapper={wrapper}
            item={item}
            next={next}
            spacer={spacer}
        />)

    expect(tree).toMatchSnapshot()
})
