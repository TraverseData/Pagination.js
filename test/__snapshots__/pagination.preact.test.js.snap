// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preact/Pagination 1st page, 10 total, 10 per page 1`] = `null`;

exports[`preact/Pagination 1st page, 100 total, 10 per page 1`] = `
<ul>
  <li>
    1
  </li>
  <li>
    <a
      href={2}
    >
      2
    </a>
  </li>
  <li>
    <a
      href={3}
    >
      3
    </a>
  </li>
  <li>
    ...
  </li>
  <li>
    <a
      href={10}
    >
      10
    </a>
  </li>
  <li>
    <a
      href={2}
    >
      Next
    </a>
  </li>
</ul>
`;

exports[`preact/Pagination 3rd page, 45 total, 10 per page 1`] = `
<ul>
  <li>
    <a
      href={2}
    >
      Previous
    </a>
  </li>
  <li>
    <a
      href={1}
    >
      1
    </a>
  </li>
  <li>
    <a
      href={2}
    >
      2
    </a>
  </li>
  <li>
    3
  </li>
  <li>
    <a
      href={4}
    >
      4
    </a>
  </li>
  <li>
    <a
      href={5}
    >
      5
    </a>
  </li>
  <li>
    <a
      href={4}
    >
      Next
    </a>
  </li>
</ul>
`;

exports[`preact/Pagination 10th page, 100 total, 10 per page 1`] = `
<ul>
  <li>
    <a
      href={9}
    >
      Previous
    </a>
  </li>
  <li>
    <a
      href={1}
    >
      1
    </a>
  </li>
  <li>
    ...
  </li>
  <li>
    <a
      href={8}
    >
      8
    </a>
  </li>
  <li>
    <a
      href={9}
    >
      9
    </a>
  </li>
  <li>
    10
  </li>
</ul>
`;

exports[`preact/Pagination 10th page, 200 total, 10 per page 1`] = `
<ul>
  <li>
    <a
      href={9}
    >
      Previous
    </a>
  </li>
  <li>
    <a
      href={1}
    >
      1
    </a>
  </li>
  <li>
    ...
  </li>
  <li>
    <a
      href={8}
    >
      8
    </a>
  </li>
  <li>
    <a
      href={9}
    >
      9
    </a>
  </li>
  <li>
    10
  </li>
  <li>
    <a
      href={11}
    >
      11
    </a>
  </li>
  <li>
    <a
      href={12}
    >
      12
    </a>
  </li>
  <li>
    ...
  </li>
  <li>
    <a
      href={20}
    >
      20
    </a>
  </li>
  <li>
    <a
      href={11}
    >
      Next
    </a>
  </li>
</ul>
`;
